src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/extensions/arquillian/ConfigureArquillianTask.groovy

```diff
@@ -1,67 +1,108 @@
 package net.wasdev.wlp.gradle.plugins.tasks.extensions.arquillian;
 
+import org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler
+import org.gradle.api.internal.tasks.TaskDependencyInternal
+import org.w3c.dom.Element
+import net.wasdev.wlp.gradle.plugins.utils.LooseWarApplication
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.apache.commons.io.FileUtils;
 import java.io.File
 import java.io.FileNotFoundException
 import java.io.IOException
 import java.util.Map.Entry
-
 import javax.xml.parsers.ParserConfigurationException
 import javax.xml.xpath.XPathExpressionException
-
 import org.gradle.api.GradleException
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.impldep.bsh.This
 import org.xml.sax.SAXException
-
+import net.wasdev.wlp.common.arquillian.objects.LibertyProperty;
+import net.wasdev.wlp.common.arquillian.objects.LibertyRemoteObject
 import net.wasdev.wlp.common.arquillian.objects.LibertyManagedObject
 import net.wasdev.wlp.common.arquillian.objects.LibertyProperty
 import net.wasdev.wlp.common.arquillian.util.ArquillianConfigurationException
 import net.wasdev.wlp.common.arquillian.util.Constants
 import net.wasdev.wlp.common.arquillian.util.HttpPortUtil
 import net.wasdev.wlp.gradle.plugins.tasks.AbstractServerTask
+import net.wasdev.wlp.gradle.plugins.tasks.extensions.arquillian.ConfigureArquillianTask.TypeProperty
 
 class ConfigureArquillianTask extends AbstractServerTask {
 
-    @Input
-    public boolean skipIfArquillianXmlExists = false;
+	Map<String, String> arquillianProperties;
+
+	public TypeProperty type = TypeProperty.NOTFOUND
+	public enum TypeProperty {
+		MANAGED, REMOTE, NOTFOUND;
+	}
+	@Input
+	public boolean skipIfArquillianXmlExists = false;
+
+	@Input
+	public Map<String, String> arquillianProperties = null;
+
+	@TaskAction
+	void doExecute() throws GradleException {
+		File arquillianXml = new File(project.getBuildDir(), "resources/test/arquillian.xml");
+		project.configurations.testCompile.each {
+
+			if(it.toString().contains("arquillian-wlp-remote")) {
+				type = TypeProperty.REMOTE
+			}
+			else if(it.toString().contains("arquillian-wlp-managed")) {
+				type = TypeProperty.MANAGED
+			}
+		}
 
-    @Input
-    public Map<String, String> arquillianProperties = null;
+		if (skipIfArquillianXmlExists && arquillianXml.exists()) {
```

Comment:
Make this logic consistent with [ConfigureArquillianMojo](https://github.com/chyt/ci.maven/blob/master/liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/extensions/arquillian/ConfigureArquillianMojo.java) in Maven. Should default to managed container. also use a `switch` statement instead of conditionals.